mesonbuild/compilers/zig.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2012-2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os.path
+import subprocess
+import textwrap
+import typing as T
+
+from .compilers import Compiler
+from ..mesonlib import MachineChoice, EnvironmentException
+
+if T.TYPE_CHECKING:
+    from ..envconfig import MachineInfo
+    from ..environment import Environment
+    from ..dependencies import ExternalProgram
+
+zig_optimization_args = {
+    '0': [],
+    'g': ['-O', 'Debug'],
+    '1': ['-O', 'ReleaseSafe'],
+    '2': ['-O', 'ReleaseSafe'],
+    '3': ['-O', 'ReleaseFast'],
+    's': ['-O', 'ReleaseSmall'],
+}  # type: T.Dict[str, T.List[str]]
+
+
+class ZigCompiler(Compiler):
+    language = 'zig'
+
+    def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoice, info: 'MachineInfo',
+                 exe_wrapper: T.Optional['ExternalProgram'] = None, is_cross: bool = False):
+        super().__init__(exelist, version, for_machine, info, is_cross=is_cross)
+        self.id = 'zig'
+        self.exe_wrapper = exe_wrapper
+
+    def get_optimization_args(self, optimization_level: str) -> T.List[str]:
+        return zig_optimization_args[optimization_level]
+
+    def get_output_args(self, outputname: str) -> T.List[str]:
+        return ['-femit-bin=%s' % outputname]
```

Comment:
I originally tried to use `--name` however it can't handle paths which causes lld to error as meson tries to invoke it with a path instead of just a filename.

Reply:
Odd, the documentation doesn't say this option takes any arguments, just that it changes the output mode. There is also `--out-dir`, so maybe we need to split the output path and pass two arguments?